blocks/library/code-mirror/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import CodeMirror from 'react-codemirror';
+require( 'codemirror/mode/css/css' );
+require( 'codemirror/mode/diff/diff' );
+require( 'codemirror/mode/elm/elm' );
+require( 'codemirror/mode/javascript/javascript' );
+require( 'codemirror/mode/markdown/markdown' );
+require( 'codemirror/mode/pegjs/pegjs' );
+require( 'codemirror/mode/php/php' );
+require( 'codemirror/lib/codemirror.css' );
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'i18n';
+import InspectorControls from '../../inspector-controls';
+import { registerBlockType, query } from '../../api';
+const { prop } = query;
+
+registerBlockType( 'core/code-mirror', {
+	title: wp.i18n.__( 'Code Editor' ),
+	icon: 'text',
+	category: 'formatting',
+
+	attributes: {
+		content: prop( 'code', 'textContent' ),
+	},
+
+	edit( { attributes, focus, setAttributes } ) {
+		return (
+			<div>
+				<CodeMirror
+					value={ attributes.content }
+					onChange={ value => setAttributes( { content: value } ) }
+					options={ {
+						lineNumbers: true,
+						mode: attributes.language,
+					} }
+				/>
+				{ focus && (
+					<InspectorControls>
+						<label className="blocks-text-control__label" htmlFor="blocks-codemirror-language-select">{ __( 'Language' ) }</label>
+						<select // eslint-disable-line jsx-a11y/no-onchange
+							id="blocks-codemirror-language-select"
+							onChange={ ( { target: { value } } ) => setAttributes( { language: value } ) }
```

Comment:
@iseulde I changed this back to `onChange` because `onBlur` wasn't updating the setting until I clicked outside of the box. now I'm in a quandary: `eslint` wants `onBlur` but it appears like we have to use `onChange` to get the highlighting to update with the selection.

any thoughts?